lib/kubernetes-deploy/container_logs.rb

```diff
@@ -50,6 +52,10 @@ def fetch_latest(kubectl)
       out.split("\n")
     end
 
+    def kubectl
+      @kubectl ||= Kubectl.new(namespace: @namespace, context: @context, logger: @logger, log_failure_by_default: false)
```

Comment:
`sync` can be called from multiple threads at once, right? and `sync` calls `fetch_latest` which calls `kubectl`, which writes an instance variable here. Probably totally harmless, but I think in theory there's a race condition here where two threads both run `kubectl` while `@kubectl` is still `nil` and then the second one will overwrite the `@kubectl` that the first one created.

Reply:
> just curious: why do we not want to log failures by default?

Because logs are a nicety as far as this gem is concerned, and they can be unavailable for reasons other than kubectl commands failing. Regardless of why we don't have them, we just print `None found. Please check your usual logging service (e.g. Splunk).`.  

There are many other cases where we also don't log kubectl errors by default, because we assume they are transient and retry on the next polling cycle. You can use `DEBUG=1` to turn on error logging regardless of which setting is used bye the `Kubectl` instance.